src/plone/restapi/services/users/get.py

```diff
@@ -92,6 +93,8 @@ def _principal_search_results(
     def _get_users(self, **kw):
         results = {user["userid"] for user in self.acl_users.searchUsers(**kw)}
         users = [self.portal_membership.getMemberById(userid) for userid in results]
+        # Filtering for None which might happen due to some unknown condition
+        users = users.filter(lambda x: x is not None)
```

Comment:
I would use generators here and above in l95 in order to reduce the memory footprint if many users are requested.

Reply:
ok in l95 but not good in l96 because the result will get in-place sorted, which is not possible with generators.

Reply:
You could use `sorted()` instead of in place sort and pass generators all way long until the result is serialized to json.